test/box/gh-6405-discarding-arguments-in-box-func-call.result

```diff
@@ -0,0 +1,190 @@
+-- test-run result file version 2
+env = require('test_run')
+ | ---
+ | ...
+test_run = env.new()
+ | ---
+ | ...
+
+test_run:cmd("create server test with script = \"box/lua/cfg_test1.lua\"")
+ | ---
+ | - true
+ | ...
+test_run:cmd("start server test")
+ | ---
+ | - true
+ | ...
+test_run:cmd("switch test")
+ | ---
+ | - true
+ | ...
+
+sum_args = function(...)                      \
+    local type_table = {}                     \
+    for i = 1, select('#', ...) do            \
+    type_table[i] = type(select(i, ...))      \
+    end                                       \
+    return type_table                         \
+end                                           \
+
+function TablesEquals(table1, table2)         \
+    for k, v in pairs(table1) do              \
+        if table2[k] ~= v then                \
+            return false                      \
+        end                                   \
+    end                                       \
+                                              \
+    for k, v in pairs(table2) do              \
+        if table1[k] ~= v then                \
+            return false                      \
+        end                                   \
+    end                                       \
+                                              \
+    return true                               \
+end                                           \
+
+box.schema.func.create('sum_args')
+ | ---
+ | ...
+
+box_call = box.func.sum_args:call({1, nil, 3})
+ | ---
+ | ...
+lua_call = sum_args(1, nil, 3)
+ | ---
+ | ...
+assert (box_call == lua_call)
+ | ---
+ | - error: assertion failed!
+ | ...
+box_call
+ | ---
+ | - - number
+ |   - nil
+ |   - number
+ | ...
+lua_call
+ | ---
+ | - - number
+ |   - nil
+ |   - number
+ | ...
+
+box_call = box.func.sum_args:call({'string', nil, 3})
+ | ---
+ | ...
+lua_call = sum_args('string', nil, 3)
+ | ---
+ | ...
+assert (TablesEquals(box_call, lua_call))
+ | ---
+ | - true
+ | ...
+box_call
+ | ---
+ | - - string
+ |   - nil
+ |   - number
+ | ...
+lua_call
+ | ---
+ | - - string
+ |   - nil
+ |   - number
+ | ...
```

Comment:
Do you need to print both tables if they are equal?

Reply:
I think this can be useful when the user has received different tables and wants to understand how they are different.

Reply:
You can print them in the case of assertion fail.

Reply:
fixed